src/openrct2/entity/Guest.cpp

```diff
@@ -2617,7 +2617,13 @@ static bool peep_check_ride_price_at_entrance(Guest* peep, Ride* ride, money32 r
 
     if (ridePrice > peep->CashInPocket)
     {
-        peep->InsertNewThought(PeepThoughtType::CantAffordRide, peep->CurrentRide);
+        // Prevent looping of same thought / animation since Destination Tolerance
+        // is only 0 exactly at entrance and will immediately change as guest
+        // tries to leave hereafter
+        if (peep->DestinationTolerance == 0)
```

Comment:
This is only changing the thought generation so ultimate does nothing different.

Reply:
This thought is paired with an animation. Since this code is continuously called, the guest became stuck in said animation so it wouldn't ever leave the queue.

This change prevents the endless loop, resolving the issue. You can test that it works with the save file as provided in the issue.